src/main/java/com/obj/nc/repositories/DeliveryInfoRepository.java

```diff
@@ -73,6 +73,42 @@ List<DeliveryInfo> findByIntentIdAndStatusOrderByProcessedOn(
     List<DeliveryInfo> findByEventIdAndEndpointIdOrderByProcessedOn(@Param("eventId") UUID eventId,
                                                                     @Param("endpointId") UUID endpointId);
 
+    String QRY_LATEST_DELIVERY_INFO_BY_ENDPOINT_ID = "WITH temp AS (\n" +
```

Comment:
kind of copy/paste this sql and the secong. 
+ 'temp' is not very descriptive name.. would suggest to name the temprary teble/view more related to what it contains